publication-commons/src/main/java/no/unit/nva/publication/service/impl/UpdateResourceService.java

```diff
@@ -357,6 +372,25 @@ private TransactWriteItem updateResource(Resource resourceUpdate) {
         return new TransactWriteItem().withPut(put);
     }
 
+    private TransactWriteItem permanentlyRemove(Resource resourceUpdate) {
+
+        var resourceDao = new ResourceDao(resourceUpdate);
+
+        var primaryKeyConditionAttributeValues = primaryKeyEqualityConditionAttributeValues(resourceDao);
+
+        var keyToDelete = new HashMap<String, AttributeValue>();
+        keyToDelete.put("PK0", new AttributeValue(resourceDao.getPrimaryKeyPartitionKey()));
+        keyToDelete.put("SK0", new AttributeValue(resourceDao.getPrimaryKeySortKey()));
+        Delete delete = new Delete()
+                            .withTableName(tableName)
+                            .withKey(keyToDelete)
+                            .withConditionExpression(PRIMARY_KEY_EQUALITY_CHECK_EXPRESSION)
+                            .withExpressionAttributeNames(PRIMARY_KEY_EQUALITY_CONDITION_ATTRIBUTE_NAMES)
+                            .withExpressionAttributeValues(primaryKeyConditionAttributeValues);
+
+        return new TransactWriteItem().withDelete(delete);
```

Comment:
Create a transactWriteItem that contains both the removal and the insertion of the new publication so that we get an atomic operation.